From 8f78817ba2018a5f76ebe78b251e2ad4b5b35457 Mon Sep 17 00:00:00 2001
From: Robert Mustacchi <rm@joyent.com>
Date: Wed, 27 Jul 2016 17:09:29 +0000
Subject: [PATCH] OS-5541 Useless assert in acpidev_cpu_query_MAT() Reviewed
 by: Jerry Jelinek <jerry.jelinek@joyent.com> Reviewed by: Joshua M. Clulow
 <jmc@joyent.com> Approved by: Joshua M. Clulow <jmc@joyent.com>

---
 usr/src/uts/i86pc/io/acpi/acpidev/acpidev_cpu.c | 2 +-
 usr/src/uts/i86pc/io/mp_platform_common.c       | 6 ++++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/usr/src/uts/i86pc/io/acpi/acpidev/acpidev_cpu.c b/usr/src/uts/i86pc/io/acpi/acpidev/acpidev_cpu.c
index a108015bbb..ebfc0fcd86 100644
--- a/usr/src/uts/i86pc/io/acpi/acpidev/acpidev_cpu.c
+++ b/usr/src/uts/i86pc/io/acpi/acpidev/acpidev_cpu.c
@@ -256,11 +256,11 @@ acpidev_cpu_query_MAT(ACPI_SUBTABLE_HEADER *ap, void *context)
 	switch (ap->Type) {
 	case ACPI_MADT_TYPE_LOCAL_APIC:
 		mpa = (ACPI_MADT_LOCAL_APIC *)ap;
-		ASSERT(mpa->Id != 255);
 		rp->found = B_TRUE;
 		rp->proc_id = mpa->ProcessorId;
 		rp->apic_id = mpa->Id;
 		if (mpa->LapicFlags & ACPI_MADT_ENABLED) {
+			ASSERT(mpa->Id != 255);
 			rp->enabled = B_TRUE;
 		} else {
 			rp->enabled = B_FALSE;
diff --git a/usr/src/uts/i86pc/io/mp_platform_common.c b/usr/src/uts/i86pc/io/mp_platform_common.c
index 51ddd5b17b..36ebd180ca 100644
--- a/usr/src/uts/i86pc/io/mp_platform_common.c
+++ b/usr/src/uts/i86pc/io/mp_platform_common.c
@@ -643,6 +643,12 @@ acpi_probe(char *modname)
 		case ACPI_MADT_TYPE_LOCAL_APIC:
 			mpa = (ACPI_MADT_LOCAL_APIC *) ap;
 			if (mpa->LapicFlags & ACPI_MADT_ENABLED) {
+				if (mpa->Id == 255) {
+					cmn_err(CE_WARN, "!%s: encountered "
+					    "invalid entry in MADT: CPU %d "
+					    "has Local APIC Id equal to 255 ",
+					    psm_name, mpa->ProcessorId);
+				}
 				if (mpa->Id == local_ids[0]) {
 					ASSERT(index == 1);
 					proc_ids[0] = mpa->ProcessorId;
-- 
2.21.0

